internal/clients/spotify/spotify.go

```diff
@@ -0,0 +1,246 @@
+package spotify
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/SongStitch/song-stitch/internal/constants"
+	"github.com/SongStitch/song-stitch/internal/models"
+	"github.com/rs/zerolog"
+)
+
+type SpotifyAuthResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"` // seconds
+}
+
+type Token struct {
+	AccessToken string
+	ExpiresIn   int
+	client      *http.Client
+	endpoint    string
+}
+
+func (t *Token) Refresh() error {
+	client_id := os.Getenv("SPOTIFY_CLIENT_ID")
+	client_secret := os.Getenv("SPOTIFY_CLIENT_SECRET")
+
+	if client_id == "" || client_secret == "" {
+		return errors.New("spotify credentials not set")
+	}
+
+	data := url.Values{}
+	data.Set("grant_type", "client_credentials")
+	data.Set("client_id", client_id)
+	data.Set("client_secret", client_secret)
+
+	req, err := http.NewRequest(http.MethodPost, t.endpoint, strings.NewReader(data.Encode()))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	res, err := t.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return errors.New("spotify authentication failed")
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	var response SpotifyAuthResponse
+	err = json.Unmarshal([]byte(body), &response)
+	if err != nil {
+		return err
+	}
+	t.AccessToken = response.AccessToken
+	t.ExpiresIn = response.ExpiresIn
+	return nil
+}
+
+func (t *Token) KeepAlive(log zerolog.Logger) {
```

Comment:
From what understand, this is running as a goroutine and it updates the token — which is scoped for other functions?